### ISSUE TITLE
Custom 404 Handler Returns HTTP 200

### ISSUE THREAD

**Issue by steeleforge** [Adobe-Marketing-Cloud/aem-project-archetype#11]
https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/blob/master/src/main/archetype/ui.apps/src/main/content/jcr_root/apps/sling/servlet/errorhandler/404.jsp

Teams using this archetype may encounter issues with this custom 404 handler as it returns HTTP 200 OK. Some hosting providers may leverage a healthcheck system or load balancing strategy which depends on a guaranteed non-existing page, and expect HTTP status code 404 to be served from a running AEM instance.

**Comment by rombert** [comment]
Sorry for the delay. I agree that the error handler should return status code 404. That's easily done in JSP, but I'd prefer if we did this using Sightly. 

Would you be willing to convert this script to sightly and submit a PR? If not, I'll do it myself next week since I want to release a new version of the archetype then.

**Comment by steeleforge** [comment]
Please do, I don't personally know if sightly is going to be resolved by sling in the same way without falling back to the default JSP found in libs. Doing this in sightly would involve a new class extending WCMUse and performing similar logic of course but may have to wrap CRLF whitespace with sightly comments if the data-sly-use is more than one line or has children tags.

**Comment by rombert** [comment]
ACK, I'll do this myself next week. The first step is simple - 404.jsp renamed to 404.html . But then I probably need a use object for response.setStatus(404).

Would that be enough to close this bug?

**Comment by steeleforge** [comment]
absolutely if it works as expected! Sorry, I would have solved this myself with sightly, but wouldn't be able to get to this for a few weeks (near a launch/go-live).

**Comment by rombert** [comment]
Don't worry, I'm happy to provide the fix.

I just wanted to give you the chance should you have the desire _and_ the
time. You're our top external contributor anyway :-)

On Fri, Feb 20, 2015 at 6:37 PM, David Steele <notifications@github.com>
wrote:

> absolutely if it works as expected! Sorry, I would have solved this myself
> with sightly, but wouldn't be able to get to this for a few weeks (near a
> launch/go-live).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/issues/11#issuecomment-75268973>
> .
>



-- 
http://robert.muntea.nu/

**Comment by rombert** [comment]
Don't worry, I'm happy to provide the fix :-)

I just wanted to give you the chance should you have the desire _and_ the
time. You're our top external contributor

absolutely if it works as expected! Sorry, I would have solved this myself
with sightly, but wouldn't be able to get to this for a few weeks (near a
launch/go-live).

—
Reply to this email directly or view it on GitHub
<https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/issues/11#issuecomment-75268973>
.

**Comment by steeleforge** [comment]
That's very kind. I'm glad to see this work take external contribution. When our CQ4 team received CQ5 we lamented the loss of supporting tooling like CRXDE despite it's many flaws and struggled to hack a CQ package XML files, manually edit OSGI MANIFEST.MF files, and make ugly ant scripts to push packages to early CQ5. Then the public maven repo and tooling ecosystem came around and we rejoiced :)
